rts/lra/test/basic/src/test/java/io/narayana/lra/arquillian/OpenAPITestIT.java

```diff
@@ -0,0 +1,73 @@
+/*
+ * JBoss, Home of Professional Open Source.
+ * Copyright 2019, Red Hat, Inc., and individual contributors
+ * as indicated by the @author tags. See the copyright.txt file in the
+ * distribution for a full listing of individual contributors.
+ *
+ * This is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as
+ * published by the Free Software Foundation; either version 2.1 of
+ * the License, or (at your option) any later version.
+ *
+ * This software is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this software; if not, write to the Free
+ * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+ */
+
+package io.narayana.lra.arquillian;
+
+import io.narayana.lra.arquillian.resource.LRAParticipantWithStatusURI;
+import io.narayana.lra.arquillian.resource.LRAParticipantWithoutStatusURI;
+import io.narayana.lra.arquillian.resource.SimpleLRAParticipant;
+import org.jboss.arquillian.container.test.api.Deployment;
+import org.jboss.arquillian.test.api.ArquillianResource;
+import org.jboss.logging.Logger;
+import org.jboss.shrinkwrap.api.spec.WebArchive;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.TestName;
+
+import javax.ws.rs.core.Response;
+import java.net.MalformedURLException;
+import java.net.URISyntaxException;
+import java.net.URL;
+
+public class OpenAPITestIT extends TestBase {
+    private static final Logger log = Logger.getLogger(OpenAPITestIT.class);
+
+    @ArquillianResource
+    public URL baseURL;
+
+    @Rule
+    public TestName testName = new TestName();
+    private URL url;
+
+    @Override
+    public void before() {
+        super.before();
+        log.info("Running test " + testName.getMethodName());
+    }
+
+    @Deployment
+    public static WebArchive deploy() {
+        return Deployer.deploy(OpenAPITestIT.class.getSimpleName(),
+                LRAParticipantWithStatusURI.class,
+                LRAParticipantWithoutStatusURI.class,
+                SimpleLRAParticipant.class);
+    }
+
+    @Test
+    public void test() throws URISyntaxException, MalformedURLException {
+        assert(coordinatorUrl.endsWith("/lra-coordinator/lra-coordinator"));
+        URL url = new URL("http://" + System.getProperty("lra.coordinator.host", "localhost")+":"+ System.getProperty("lra.coordinator.port", "8080")+"/openapi");
+        Response response = client.target(url.toURI()).request().get();
+        String output = response.readEntity(String.class);
+        assert(output.contains("/lra-coordinator/lra-coordinator"));
```

Comment:
The test fails as this is not now part of the openAPI output. Also, why not use Junit Assert or Assertions (not sure which version we use here) with a more descriptive error message?

Reply:
I don't quite understand how I missed it when preparing the test but the test is failing here on WildFly. I think this could be a problem then. The reason I say that is the endpoint will be different with this change so it's a public API change. I thought WildFly would stay on "/lra-coordinator/lra-coordinator". I am going to put a hold here.

Reply:
I am finding that for WildFly the coordinator is indeed on /lra-coordinator/lra-coordinator but the openapi is now showing /lra-coordinator which I was not expecting. I do understand from the change what might cause this but it's not what I would expect we need.

Reply:
I think my addition of https://github.com/jbosstm/narayana/blob/eed7548e37ed5a915ba92bdd8cc61c880605a39c/rts/lra/coordinator/src/main/java/io/narayana/lra/coordinator/api/Coordinator.java#L129 has changed this, at least from WildFly point of view. I need to find out what is expected when a class has an `@Path` and `@ApplicationPath`

Reply:
I probably thought I was using JUnit assert - will change that - thanks for mentioning.

Reply:
Done, thanks